core/rawdb/accessors_chain.go

```diff
@@ -1193,7 +1193,7 @@ func Transitioned(db kv.Getter, blockNum uint64) (trans bool, err error) {
 	if len(data) == 0 {
 		return false, nil
 	}
-	return blockNum > binary.BigEndian.Uint64(data), nil
+	return blockNum >= binary.BigEndian.Uint64(data), nil
```

Comment:
I'm afraid we cannot rely on the block number for determining transition to PoS. The reason is that it's defined in terms of the total difficulty; and theoretically [Terminal PoW blocks](https://eips.ethereum.org/EIPS/eip-3675#definitions) (which can be several) might have different block heights.